lib/acts_as_ar_model.rb

```diff
@@ -142,8 +142,30 @@ def self.reflections
   # Find routines
   #
 
+  # This method is called by QueryRelation upon executing the query.
+  #   Since it will receive non-legacy search options, we need to convert
+  #   them to handle the legacy find.
+  def self.search(mode, options = {})
```

Comment:
good news: you won me over on using `search`
bad news: now I'm not sure if I want to link `klass` and `search` together.

oh, and did I say how much I liked how you pulled out `to_legacy_options`? very cools tuff